### PR TITLE
[Bug] Fix #762: All Pokemon become invisible when capturing then switching with your only pokemon that was not fainted

### LINKED ISSUE #762: All Pokemon become invisible when capturing then switching with your only pokemon that was not fainted.
Was against castform Boss, with my Patrat who was the only one alive. I captured the castform and switch it for my patrat and now all Pokemon are invisible in future challenges.
image

### PR DESCRIPTION
**Pull request by flx-sta**
## What are the changes the user will see?

All sprites will no longer disappear if the only allowed Pokemon in the party is replaced with the newly caught one.

## Why am I making these changes?
fixes #762 

## What are the changes from a developer perspective?
Right now when a Pokemon is added to a party it's always simply `push()`ed to the end. 
I've simply added an optional `slotIndex` that will put a newly caught Pokemon at the exact position where the previous one was released. That fixes it too 

### Replace only legal PKM
<details>
    <summary>⏯️ Video: Replace only legal PKM</summary>
    
https://github.com/user-attachments/assets/b6092c00-4333-4023-8674-50aa6a274122

</details>

### Replace 2nd active in double battle

<details>
    <summary>⏯️ Video: Replace 2nd active in double battle</summary>
    
https://github.com/user-attachments/assets/30e2c974-fa34-4963-9f89-fb256eedecba

</details>

### Replace with any fainted in party

<details>
    <summary>⏯️ Video: Replace with any fainted in party</summary>
    
https://github.com/user-attachments/assets/b19f187e-274b-43d3-80a9-e63cc261dd4c

</details>

### Catch and add since party isn't full

<details>
    <summary>⏯️ Video: Catch and add since party isn't full</summary>
    
https://github.com/user-attachments/assets/b5381f30-d595-41b4-8bd8-9f90e39abdfa

</details>

## How to test the changes?
Follow the instructions in #762 !

### Overrides
```ts
const overrides = {
  POKEBALL_OVERRIDE:  {
    active: true,
    pokeballs: {
      [PokeballType.POKEBALL]: 99,
      [PokeballType.GREAT_BALL]: 99,
      [PokeballType.ULTRA_BALL]: 99,
      [PokeballType.ROGUE_BALL]: 99,
      [PokeballType.MASTER_BALL]: 99,
    },
  },
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```

### Session Save-file
[762-testing-session.prsv](https://github.com/user-attachments/files/16880093/762-testing-session.txt)
<sup>Replace file-extension with `.prsv` to allow import</sup>

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~~[] If I have text, did I make it translatable and add a key in the English locale file(s)?~~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const PLAYER_PARTY_MAX_SIZE = 6;
```
Comment (Tempo-anon) [inline src/constants.ts:1]: Getting fancy are we?

File: src/test/field/pokemon.test.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { Species } from "#app/enums/species";
 import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
 import GameManager from "../utils/gameManager";
+import { PokeballType } from "#app/enums/pokeball.js";
+import BattleScene from "#app/battle-scene.js";
```
Comment (Tempo-anon) [inline src/test/field/pokemon.test.ts:5]: nit: js imports

File: src/utils.ts [hunk 3]
```diff
@@ -583,3 +583,14 @@ export function toDmgValue(value: number, minValue: number = 1) {
...
 }
+
+/**
+ * Check if a number is between two other numbers
+ * @param num the number to check
+ * @param min the minimum value
+ * @param max the maximum value
+ * @returns true if number is between min and max
+ */
+export function isBetween(num: number, min: number, max: number): boolean {
```
Comment (Tempo-anon) [inline src/utils.ts:594]: Should probably mention the check is inclusive

File: src/constants.ts (hunk 1 above)
Comment (flx-sta) [reply src/constants.ts:1]: Of course, sir

File: src/utils.ts (hunk 3 above)
Comment (flx-sta) [reply src/utils.ts:594]: Will do

**Comment by flx-sta** [comment]
@Tempo-anon Please re-review

**Review by frutescens — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**